### PR TITLE
Add a MaximumRowsOrColumns to Uniformgridlayout

### PR DESCRIPTION
**Pull request by micahl**
I had a small bit of down-time on my break and took a crack at adding the MaximumRowsOrColumns property.  Involved a few more changes than I initially thought it would.

### CODE REVIEW

File: dev/Repeater/FlowLayout.cpp [hunk 1]
```diff
@@ -57,6 +57,7 @@ winrt::Size FlowLayout::MeasureOverride(
         true, /* isWrapping*/
         MinItemSpacing(),
         LineSpacing(),
+        MAXUINT /* maxItemsPerLine */,
```
Comment (ranjeshj) [inline dev/Repeater/FlowLayout.cpp]: nit: std::numeric_limits<int>::max()

File: dev/Repeater/UniformGridLayoutState.cpp [hunk 2]
```diff
@@ -36,15 +36,21 @@ void UniformGridLayoutState::EnsureElementSize(
     const winrt::UniformGridLayoutItemsStretch& stretch,
     const winrt::Orientation& orientation,
     double minRowSpacing,
-    double minColumnSpacing)
+    double minColumnSpacing,
+    unsigned int maxItemsPerLine)
 {
+    if (maxItemsPerLine == 0)
```
Comment (ranjeshj) [inline dev/Repeater/UniformGridLayoutState.cpp]: Should we perhaps throw in this case instead of forcing it be 1 ?

Comment (micahl) [reply dev/Repeater/UniformGridLayoutState.cpp]: We could.  I was trying to mimic the behavior of WrapGrid (and I think) ItemsWrapGrid where it doesn't complain about 0 and still shows at least 1.

**Comment by ranjeshj** [comment]
azp /run

**Pull request merged**